messaging/src/main/java/org/axonframework/common/ConstructorUtils.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.common;
+
+import jakarta.annotation.Nonnull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Utility class for constructing objects using reflection.
+ * <p>
+ * Utility functions in this class do not create instances directly. Instead, they provide reusable functions that can
+ * be used to construct a multitude of instances of the same type. It is advisable to cache this function for reuse.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class ConstructorUtils {
```

Comment:
Some tests for this new utility class would be nice. Although the tests are potentially cumbersome, as this util class might become a cornerstone of a lot of other code, it should be validated thoroughly.

Reply:
Thank you! :pray: